### ISSUE TITLE
Continuous benchmarking

### ISSUE THREAD

**Issue by tarcieri** [RustCrypto/actions#30]
We've had a couple notable performance regressions lately, namely in [`keccak`](https://github.com/RustCrypto/hashes/issues/468) and [`p256`](https://github.com/RustCrypto/elliptic-curves/issues/882).

It'd be good to have a reusable solution for continuous benchmarking.

Some options:

- https://github.com/bencherdev/bencher
- https://github.com/bheisler/iai

**Comment by Slixe** [comment]
Criterion may be a good alternative too:
- https://github.com/bheisler/criterion.rs

**Comment by tarcieri** [comment]
Yeah, we already use criterion extensively.

For this particular problem though, we need to be able to produce reliable benchmarking results in a virtualized CI environment, which is not something criterion provides a solution for.

That particular problem is what makes something like `iai` interesting.

**Comment by Slixe** [comment]
Agree, `iai` looks promising, and the perfect choice for benchmarks of hashing algorithms.
